### ISSUE TITLE
Upload releases in parallel

### ISSUE THREAD

**Issue by apoydence** [cloudfoundry/bosh-cli#260]
While deploying [cf-deployment](https://github.com/cloudfoundry/cf-deployment) for the first time, bosh goes and downloads several releases. It does so one at a time. I believe it would be faster to download several releases at once instead.

I've timed deploying cf-deployment on my local machine:
```
317.94 real         3.27 user         0.73 sys
```

I then updated the bosh-cli (as a spike) to download 5 releases at a time and deployed to a new local director:
```
231.51 real         6.07 user         1.70 sys
```

This comes to ~2 minutes quicker.

I would be happy doing a PR if this is reasonable.

**Comment by dpb587-pivotal** [comment]
Agreed - closing in favor of #221 and pending PR #229.
